### PR TITLE
feat!: Open Food Facts website redesign

### PR DESCRIPTION
**Pull request by stephanegigandet**
This PR builds on the work of @fredDesign to implement the new design created by Quentin for the OFF website (available in Figma: https://www.figma.com/file/Qg9URUyrjHgYmnDHXRsTTB/New-website-design-(2022)-(Quentin)?node-id=0%3A1

It's not completely finishes, but I think it's almost ready to reploy. Remaining issues are tracked in this bug: https://github.com/openfoodfacts/openfoodfacts-server/issues/7413

image

### CODE REVIEW

**Comment by teolemon** [comment]
po/common/en.po:6279: duplicate message definition...
po/common/en.po:3910: ...this is the location of the first definition
msgfmt: found 1 fatal error

File: po/common/en.po [hunk 1]
```diff
@@ -6263,5 +6263,74 @@ msgctxt "nutrient_recommendation_fat_and_saturated_fat"
...
+msgid "Sign out"
+msgstr "Sign out"
+
+msgctxt "your_contributions"
+msgid "Your contributions"
+msgstr "Your contributions"
+
+msgctxt "products_added"
+msgid "Products added"
+msgstr "Products added"
```
Comment (teolemon) [inline po/common/en.po:6280]:
```suggestion
```

File: po/common/common.pot [hunk 2]
```diff
@@ -6258,3 +6262,75 @@ msgstr "Limit the consumption of fat and saturated fat"
...
+msgid "Sign out"
+msgstr "Sign out"
+
+msgctxt "your_contributions"
+msgid "Your contributions"
+msgstr "Your contributions"
+
+msgctxt "products_added"
+msgid "Products added"
+msgstr "Products added"
```
Comment (teolemon) [inline po/common/common.pot:6280]:
```suggestion
```

**Comment by stephanegigandet** [comment]
There are tons of eslint errors for those files, I'm putting them in .eslintignore

html/js/hc-sticky.js
html/js/scrollNav.js
html/js/stikelem.js

File: html/js/product-multilingual.js [hunk 3]
```diff
@@ -41,1076 +41,1077 @@ var imagefield_url = {};
...
+    id = id.replace("_label", "");
+    $('#' + id).focus();
+    $('#' + id + '_unit').val(ui.item.unit);
+    var unit = (ui.item.unit == '%' ? '%' : ui.item.unit).toLowerCase();
+    var unitElement = $('#' + id + '_unit');
+    var percentElement = $('#' + id + '_unit_percent');
+    if (unit === '') {
+        unitElement.hide();
+        percentElement.hide();
+    } else if (unit == '%') {
```
Comment (alexgarel) [inline html/js/product-multilingual.js:142]: It seems to me you don't use this file any more any where. Should we keep it ?

File: html/js/scrollNav.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        }
+    }, true)
+
+    /**
+     * Scroll with ofset on page load with hash links in the url
+     */
+    window.addEventListener('load', () => {
+        if (window.location.hash) {
+            if (select(window.location.hash)) {
+                scrollto(window.location.hash)
```
Comment (alexgarel) [inline html/js/scrollNav.js:134]:
```suggestion
                setTimeout(scrollto, 10, window.location.hash)
```

may resolve https://github.com/openfoodfacts/openfoodfacts-server/issues/7415

File: lib/ProductOpener/Display.pm [hunk 5]
```diff
@@ -4688,18 +4691,32 @@ sub add_params_to_query($request_ref, $query_ref) {
...
 	# Activate physical activity knowledge panel only when specified
 	if (single_param("activate_knowledge_panel_physical_activities")) {
 		$knowledge_panels_options_ref->{activate_knowledge_panel_physical_activities} = 1;
 	}
+
+	# Specify if we knowledge panels are requested from the app or the website
+	# in order to allow different behaviours (e.g. showing ingredients before nutrition on the web)
+	# possible values: "web", "app"
+	my $knowledge_panels_client = single_param("knowledge_panels_client");
+	if ((not defined $knowledge_panels_client) or (($knowledge_panels_client ne "web")) and ($knowledge_panels_client ne "app")) {
```
Comment (alexgarel) [inline lib/ProductOpener/Display.pm:4711]: reading this conditions make my head hurts :-D

I'm not sure parenthesis are set the way you want !
```suggestion
       # set a default value if client is not defined to app or web
	if ((not defined $knowledge_panels_client) or (($knowledge_panels_client ne "web") and ($knowledge_panels_client ne "app"))) {
```

File: lib/ProductOpener/Display.pm [hunk 6]
```diff
@@ -4688,18 +4691,32 @@ sub add_params_to_query($request_ref, $query_ref) {
...
+		# Default to app mode
+		$knowledge_panels_client = 'app';
+		if (not defined $request_ref->{api}) {
+			$knowledge_panels_client = "web";
+		}
```
Comment (alexgarel) [inline lib/ProductOpener/Display.pm:4716]:
```suggestion
                # but if it's not an api request, we consider it should be web
		if (not defined $request_ref->{api}) {
			$knowledge_panels_client = "web";
		}
```

File: lib/ProductOpener/Images.pm [hunk 7]
```diff
@@ -1454,6 +1454,8 @@ sub _set_magickal_options($magick, $width) {
 	return;
 }
 
+
+# This function should be removed once we switch to knowledge pages to display
```
Comment (alexgarel) [inline lib/ProductOpener/Images.pm:1458]: It's better to have a clear prefix for those kind of actions :-) 
```suggestion
# TODO: This function should be removed once we switch to knowledge pages to display
```

File: templates/web/common/includes/display_data_quality_issues_and_improvement_opportunities.tt.html [hunk 8]
```diff
@@ -1,7 +1,7 @@
 <!-- start templates/[% template.name %] -->
 
 [% FOREACH tagtype IN tagtypes %]
-
+aaa
```
Comment (alexgarel) [inline templates/web/common/includes/display_data_quality_issues_and_improvement_opportunities.tt.html:4]: did you forgot this ?

**Review by alexgarel — APPROVED** [review]
Great ! Only minor comment and one possible minor bug.

File: templates/web/common/includes/display_data_quality_issues_and_improvement_opportunities.tt.html (hunk 8 above)
Comment (stephanegigandet) [reply templates/web/common/includes/display_data_quality_issues_and_improvement_opportunities.tt.html:4]: oops, thanks!

File: lib/ProductOpener/Display.pm (hunk 5 above)
Comment (stephanegigandet) [reply lib/ProductOpener/Display.pm:4711]: This is better indeed, thanks

File: html/js/product-multilingual.js (hunk 3 above)
Comment (stephanegigandet) [reply html/js/product-multilingual.js:142]: We still use it, it's loaded by /cgi/product.pl

**Review by alexgarel — APPROVED** [review]

**Pull request merged**